python/ray/workflow/workflow_executor.py

```diff
@@ -0,0 +1,166 @@
+            future = asyncio.wrap_future(metadata_ref.future())
+            future.add_done_callback(completion_queue.put_nowait)
+
+            state.insert_running_frontier(future, WorkflowRef(task_id, ref=output_ref))
+            state.task_execution_metadata[task_id] = TaskExecutionMetadata(
+                submit_time=time.time()
+            )
+            if task_id in state.continuation_root:
+                if state.task_options[task_id].checkpoint:
+                    wf_store.update_continuation_output_link(
+                        state.continuation_root[task_id], task_id
+                    )
+            else:
+                # update reference counting if the task is not a continuation
+                for c in state.in_dependency[task_id]:
+                    state.reference_set[c].remove(task_id)
+                    if not state.reference_set[c]:
+                        del state.reference_set[c]
+                        state.free_outputs.add(c)
+
+        while state.free_outputs:
+            # garbage collect all free outputs immediately
+            gc_task_id = state.free_outputs.pop()
+            assert state.get_input(gc_task_id) is not None
+            state.output_map.pop(gc_task_id, None)
+
+        # ------------ poll ready tasks ------------
+        ready_futures = []
+        try:
+            rf = await asyncio.wait_for(completion_queue.get(), timeout=1)
+            ready_futures.append(rf)
+            # get all remaining futures in the queue
+            while not completion_queue.empty():
+                ready_futures.append(completion_queue.get_nowait())
+        except asyncio.TimeoutError:
+            continue
+
+        finish_time = time.time()
+
+        for fut in ready_futures:
+            if state.stopping:
+                raise WorkflowCancellationError(workflow_id)
+            output_ref = state.pop_running_frontier(fut)
+            task_id = output_ref.task_id
+            try:
+                metadata: WorkflowExecutionMetadata = fut.result()
+                state.task_execution_metadata[task_id].finish_time = finish_time
+                logger.info(
+                    f"Task status [{WorkflowStatus.SUCCESSFUL}]\t"
+                    f"[{workflow_id}@{task_id}]"
+                )
+            except Exception as e:
+                if isinstance(e, RayTaskError):
+                    reason = "an exception raised by the task"
+                elif isinstance(e, RayError):
+                    reason = "a system error"
+                else:
+                    reason = "an unknown error"
+                logger.error(
+                    f"Task status [{WorkflowStatus.FAILED}] due to {reason}.\t"
+                    f"[{workflow_id}@{task_id}]"
+                )
+                raise
+
+            if metadata.is_output_workflow:  # The task returns a continuation
+                sub_workflow_state: WorkflowExecutionState = await output_ref.ref
```

Comment:
Is it possible that in case of continuation, the stopping won't be executed until the await returns?